TeamCode/src/main/java/com/acmerobotics/relicrecovery/motion/MotionProfileGenerator.java

```diff
@@ -0,0 +1,202 @@
+package com.acmerobotics.relicrecovery.motion;
+
+import com.acmerobotics.relicrecovery.util.SuperArrayList;
+
+import java.util.ArrayList;
+
+/**
+ * @author kellyrm
+ *
+ * used to generate jerk-controlled seven-segment motion profiles given start and goal conditions, and some constraints
+ *
+ * like this:
+ *        1   2   3      4        5   6   7
+ *       ___                             ___
+ * J:   |   |___     ___________     ___|   |
+ *              |___|           |___|
+ *
+ * see <a href="https://pub.uni-bielefeld.de/download/1996788/2280502">this paper</a>
+ * someone should read page 4 and see if they have any clue for a implementation of the last step of the algorithm they outline
+ * what I have now is more a hack than anything
+ */
+
+public class MotionProfileGenerator {
+
+    //static class
+    private MotionProfileGenerator() {}
+
+    /**
+     * make the motion profile!
+     * @param start the initial state
+     * @param goal the goal of the profile
+     * @param constraints constraints the profile is subject to
+     * @return the motion profile!
+     */
+    public static MotionProfile generateProfile(MotionState start, MotionGoal goal, MotionConstraints constraints) {
+
+        //for simplicity, assume that the profile requires positive movement
+        if (goal.pos < start.x) {
+            //flip the profile and then flip the results
+            return generateFlippedProfile(start, goal, constraints);
+        }
+
+        MotionProfile profile = new MotionProfile(start);
+
+        //if we are headed away from the goal, first come to a stop
```

Comment:
this could be optomized, right now it dejerks, and then has to rejerk once it heads back to the goal, it would be better to make it one continous acceleration, but it still works at least

Reply:
Yeah this whole procedure is quite interesting (i.e. confusing). I read the paper, and I think I understand most of it. We might want to reimplement this directly from the paper at some point.

Reply:
Yeah, it all makes a fair ammount of sense, the one part that I could not figgure out was the last bit of the algorithm they outlined where it said something like "produce those seven equations and solve them numericaly"